static/js/formatters-internal.js

```diff
@@ -311,164 +317,122 @@ export function image(simpleOrComplexImage = {}, desiredSize = '200x', atLeastAs
   }
 
-  const isEuImage = image.url.includes('eu.mktgcdn.com');
+  let url;
+  try {
+    url = new URL(image.url);
+    let urlPath = url.pathname;
+    if (url.pathname.match('^\/p')) {
+      urlPath = _removePhotoImageUrlExtension(urlPath);
+    }
 
-  const dynamicUrl = isEuImage
-    ? _getEuImageDynamicUrl(image, desiredSize, atLeastAsLarge)
-    : _getUsImageDynamicUrl(image.url, desiredSize, atLeastAsLarge);
+    const hostname = url.hostname.replace(/^(a|dynl|dynm)\./, 'dyn.');
+    const formatOptionsString = _getImageFormatOptions(desiredSize, atLeastAsLarge, image.width, image.height);
 
-  return Object.assign(
-    {},
-    image,
-    {
-      url: dynamicUrl.replace('http://', 'https://')
-    }
-  );
+    return Object.assign(
+      {},
+      image,
+      {
+        url: `https://${hostname}${urlPath}${formatOptionsString}`,
+      }
+    );
+
+  } catch (error) {
+    throw new Error(`Error processing image url ${image.url}: ${error}`);
+  }
 }
 
 /**
- * Given a US image url, returns the dynamic url.
+ * Construct the format options string with given parameters.
  *
- * @param {string} imageUrl Image's url (e.g.
- *   'https://dynl.mktgcdn.com/p/ldMLwj1JkN94-2pwh6CjR_OMy4KnexHJCfZhPAZCbi0/196x400.jpg',
- *   'https://a.mktgcdn.com/p/ldMLwj1JkN94-2pwh6CjR_OMy4KnexHJCfZhPAZCbi0/196x400.jpg')
  * @param {string} desiredSize The desired size of the image ('<width>x<height>')
  * @param {boolean} atLeastAsLarge Whether the image should be at least as large as the desired
- *                                 size in one dimension or smaller than the desired size in both
- *                                 dimensions.
- * @returns {string} A dynamic url (e.g. 'https://dynl.mktgcdn.com/p/ldMLwj1JkN94-2pwh6CjR_OMy4KnexHJCfZhPAZCbi0/200x1.jpg')
+ *                                 size in both dimensions or at least one dimension.
+ * @param {number?} fullSizeWidth The full size width of the original image if exists
+ * @param {number?} fullSizeHeight The full size height of the original image if exists
+ * @returns {string} A string representing the format options
+ *                   (e.g. 'height=200,width=100,fit=contain')
  */
-function _getUsImageDynamicUrl(imageUrl, desiredSize, atLeastAsLarge) {
-  const [urlWithoutExtension, extension] = _splitStringOnIndex(imageUrl, imageUrl.lastIndexOf('.'));
-  const [urlBeforeDimensions, dimensions] = _splitStringOnIndex(urlWithoutExtension, urlWithoutExtension.lastIndexOf('/') + 1);
-  const fullSizeDims = dimensions.split('x');
-
-  let desiredWidth, desiredHeight;
+function _getImageFormatOptions(desiredSize, atLeastAsLarge, fullSizeWidth, fullSizeHeight) {
   let desiredDims = desiredSize.split('x');
+  const hasDesiredWidth = desiredDims[0] !== '';
+  const hasDesiredHeight = desiredDims[1] !== '';
 
-  if (desiredDims[0] !== '') {
+  // both dimensions are not provided, return original image
+  if (!hasDesiredWidth && !hasDesiredHeight) {
```

Comment:
can we update the logic in this function to follow a simplified flow like this? (apologies for the rough pseudocode)
```
const formatOptions = ['fit=contain'];
if desiredWidth && desiredHeight && originalRatio
    const mathFunc = atLeastAsLarge ? Math.max : Math.min
    use mathFunc and originalRatio to push width and height to formatOptions
    return joined formatOptions
if desiredWidth
    push width to formatOptions
if desiredHeight
    push height to formatOptions
return joined formatOptions
```
right now, having all the different cases is a little confusing to follow, not to mention that returning early like this without specifying any param would actually cause dyn to throw an error per Jacob's comment [here](https://github.com/yext/answers-hitchhiker-theme/pull/1184#discussion_r1577876311)

Reply:
I kept parsing the desiredWidth/Height number and short circuiting to be clearer. Also, I tested that without any options, the original image would be returned. Done